core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -790,8 +790,11 @@ protected LogContext getLogContext() {
         }
 
         public void run() {
-            if (isCancelled())
+            if (isCancelled() || !remoteLogMetadataManager.isReady(topicIdPartition)) {
+                logger.debug("Skipping the current run for tpId {} as it is either cancelled: {} or " +
+                        "remote log metadata is not ready", topicIdPartition, isCancelled());
```

Comment:
nit: Good to add a specific value for `!remoteLogMetadataManager.isReady(topicIdPartition)` in the statement. One way to know with this change is that if cancelled is true then this value should be true for now. It is better to explicitly add that as we may have more changes in the condition.